torch/amp/grad_scaler.py

```diff
@@ -281,6 +281,15 @@ def _unscale_grads_(
                         per_device_found_inf.get(device),
                         per_device_inv_scale.get(device),
                     )
+                    if "DTensor" in grads[0].__class__.__name__:
```

Comment:
How come this grad_scaler patch is still necessary?

Reply:
They are the necessary steps for found_inf is grad_scaler is DTensor, codes under the if condition switch `found_inf` as Tensor with inf info at all devices
If the grad_scaler is Tensor, then the code under if condition are unnecessary.

Reply:
Jane is probably asking why we need if-else base on DTensor instead of going through DTensor dispatch